grails-app/controllers/org/pih/warehouse/api/PurchaseOrderApiController.groovy

```diff
@@ -234,7 +237,6 @@ class PurchaseOrderApiController {
     }
 
     def orderItemsDerivedStatus = {
-        Order order = Order.get(params.id)
-        render order?.getOrderItemsDerivedStatus() as JSON
+        render orderSummaryService.getOrderItemsDerivedStatus(params.id) as JSON
```

Comment:
is this where the time save comes in? I'm curious why it's faster since Order.getOrderItemsDerivedStatus() calls OrderIttemSummary.findAllByOrder(this) which I would think just does a select on order id on the pre-built view.

Is the view not fully materialized at this point or something? Or is the select not indexed properly (is that even a thing on materialized views??)

Reply:
@EWaterman if I'm not mistaken, the views are recreated whenever you access them, hence this might be why it is slower, but this needs a confirmation, as I didn't check it, I just remember a theoretical statement about views.

Reply:
thanks for the explanation. I definitely agree that these flows would benefit (at least visually) from being broken down a bit. Stacking views on top of views makes me uncomfortable